src/app/modules/submission/__tests__/submission.controller.spec.ts

```diff
@@ -777,144 +401,437 @@ describe('encrypt-submission.controller', () => {
+      expect(mockRes.status).toHaveBeenCalledWith(500)
+      expect(MockSubService.getSubmissionMetadataList).not.toHaveBeenCalled()
+      expect(MockSubService.getSubmissionMetadata).toHaveBeenCalledWith(
+        FormResponseMode.Encrypt,
+        MOCK_FORM_ID,
+        mockSubmissionId,
+      )
+    })
+
+    it('should return 500 when database errors occurs for metadata list retrieval', async () => {
+      // Arrange
+      const mockReq = expressHandler.mockRequest({
+        params: { formId: MOCK_FORM_ID },
+        query: {
+          page: 1,
+        },
+        session: {
+          user: {
+            _id: MOCK_USER_ID,
+          },
+        },
+      })
+      const mockErrorString = 'error retrieving metadata list'
+      const mockRes = expressHandler.mockResponse()
+      // Mock service result.
+      MockSubService.getSubmissionMetadataList.mockReturnValueOnce(
+        errAsync(new DatabaseError(mockErrorString)),
+      )
+
+      // Act
+      await getMetadata(mockReq, mockRes, jest.fn())
+
+      // Assert
+      expect(mockRes.json).toHaveBeenCalledWith({
+        message: mockErrorString,
+      })
+      expect(mockRes.status).toHaveBeenCalledWith(500)
+      expect(MockSubService.getSubmissionMetadata).not.toHaveBeenCalled()
+      expect(MockSubService.getSubmissionMetadataList).toHaveBeenCalledWith(
+        FormResponseMode.Encrypt,
+        MOCK_FORM_ID,
+        mockReq.query.page,
+      )
+    })
+  })
+
+  describe('handleGetEncryptedResponse', () => {
+    const MOCK_FORM_ID = new ObjectId().toHexString()
+    const MOCK_USER_ID = new ObjectId().toHexString()
+
+    const MOCK_USER = {
+      _id: MOCK_USER_ID,
+      email: 'somerandom@example.com',
+    } as IPopulatedUser
+    const MOCK_FORM = {
+      admin: MOCK_USER,
+      _id: MOCK_FORM_ID,
+      title: 'mock title',
+      responseMode: FormResponseMode.Encrypt,
+    } as IPopulatedForm
+
+    const MOCK_REQ = expressHandler.mockRequest({
+      params: { formId: MOCK_FORM_ID, submissionId: 'mockSubmissionId' },
+      session: {
+        cookie: {
+          maxAge: 20000,
+        },
+        user: {
+          _id: MOCK_USER_ID,
+        },
+      },
+    })
+
+    beforeEach(() => {
+      MockUserService.getPopulatedUserById.mockReturnValue(okAsync(MOCK_USER))
+      MockAuthService.getFormAfterPermissionChecks.mockReturnValue(
+        okAsync(MOCK_FORM),
+      )
+      jest
+        .spyOn(SubmissionsUtils, 'checkFormIsEncryptModeOrMultirespondent')
+        .mockReturnValue(ok(MOCK_FORM as IPopulatedEncryptedForm))
+    })
+
+    it('should return 200 with encrypted response', async () => {
+      // Arrange
+      const mockSubData: SubmissionData = {
+        _id: 'some id',
+        encryptedContent: 'some encrypted content',
+        verifiedContent: 'some verified content',
+        created: new Date('2020-10-10'),
+        paymentId: 'payment id',
+        submissionType: SubmissionType.Encrypt,
+      } as SubmissionData
```

Comment:
Should be `as StorageModeSubmissionData` otherwise line 537 will throw a type error since `verifiedContent` does not exist on `MultirespondentSubmissionData`